pipelines/gitops-pull-request-rhtap/gitops-pull-request.yaml

```diff
@@ -80,12 +104,66 @@ spec:
           value: $(params.ec-rekor-host)
         - name: TUF_MIRROR
           value: $(params.ec-tuf-mirror)
+      runAfter:
+        - get-images-to-verify
+      taskRef:
+        name: verify-enterprise-contract
+        version: "0.1"
+
+    - name: get-images-to-upload-sbom
       runAfter:
         - gather-deploy-images
+      params:
+        - name: INPUT
+          value: $(tasks.gather-deploy-images.results.IMAGES_TO_VERIFY)
+        - name: JQ_ARGS
+          value:
+            - --compact-output
+            # upload SBOMs for images that get promoted to staging or production
+            - .components | map(select(.env != "development") | .containerImage) | unique
+        - name: RETURN_ARRAY
+          value: "true"
       when:
         - input: $(tasks.gather-deploy-images.results.IMAGES_TO_VERIFY)
           operator: notin
           values: [""]
       taskRef:
-        name: verify-enterprise-contract
+        name: jq
+        version: "0.1"
+
+    - name: download-sboms
```

Comment:
Does it make sense for this pipeline (including potential future use cases) to download sboms and do something else then upload to TPA?
Maybe we can have one huge task that would do everything? That would make the pipeline more readable and less fragile.

Reply:
If we get an upload task from the TPA team, I imagine it would be a generic task that uploads some SBOMS to TPA. We would probably still need a separate task that downloads the SBOMs and passes them to the TPA task.

In other words, I tried to keep the upload task fairly generic (not hardcode any RHTAP-isms) so it was necessary to have a separate download task.

But if I can get rid of the `jq`, the pipeline might look like this

```
              download-sboms -- upload-sboms
             /
gather-images
             \
              verify-enterprise-contract
```